client/js/components/user/CustomerSettings/CustomerSettingsContact.vue

```diff
@@ -0,0 +1,226 @@
+            }" />
+        </div>
+      </template>
+    </dp-editable-list>
+  </div>
+</template>
+
+<script>
+import { DpCheckbox, DpEditableList, DpEditor, DpInput, dpValidateMixin } from '@demos-europe/demosplan-ui'
+import { mapActions, mapMutations, mapState } from 'vuex'
+
+export default {
+  name: 'CustomerSettingsContact',
+
+  components: {
+    DpCheckbox,
+    DpEditableList,
+    DpEditor,
+    DpInput
+  },
+
+  mixins: [dpValidateMixin],
+
+  data () {
+    return {
+      customerContact: {
+        title: '',
+        phoneNumber: '',
+        eMailAddress: '',
+        text: '',
+        visible: false
+      },
+      showContactForm: false,
+      translationKeys: {
+        new: Translator.trans('customer.contact.new'),
+        add: Translator.trans('customer.contact.add'),
+        abort: Translator.trans('abort'),
+        update: Translator.trans('customer.contact.update'),
+        noEntries: Translator.trans('customer.contact.no'),
+        delete: Translator.trans('customer.contact.delete')
+      }
+    }
+  },
+
+  computed: {
+    ...mapState('customerContact', {
+      contacts: 'items'
+    })
+  },
+
+  methods: {
+    ...mapActions('customerContact', {
+      createContact: 'create',
+      fetchContacts: 'list',
+      deleteContact: 'delete',
+      saveContact: 'save'
+    }),
+
+    ...mapMutations('customerContact', {
+      updateContact: 'setItem'
+    }),
+
+    reset () {
+      this.showContactForm = false
+    },
+
+    createOrUpdateContact (id) {
+      if (id === 'new') {
+        const payload = {
+          type: 'CustomerContact',
+          attributes: {
+            title: this.customerContact.title,
+            phoneNumber: this.customerContact.phoneNumber ? this.customerContact.phoneNumber : null,
+            text: this.customerContact.text ? this.customerContact.text : null,
+            visible: this.customerContact.visible,
+            eMailAddress: this.customerContact.eMailAddress ? this.customerContact.eMailAddress : null
+          }
+        }
+        this.createContact(payload).then((response) => {
+          this.getContacts()
+          dplan.notify.notify('confirm', Translator.trans('confirm.saved'))
+        })
+      } else {
+        const payload = {
+          id: id,
+          type: 'CustomerContact',
+          attributes: {
+            title: this.customerContact.title,
+            phoneNumber: this.customerContact.phoneNumber ? this.customerContact.phoneNumber : null,
+            text: this.customerContact.text ? this.customerContact.text : null,
+            visible: this.customerContact.visible,
+            eMailAddress: this.customerContact.eMailAddress ? this.customerContact.eMailAddress : null
+          }
+        }
+        this.updateContact(payload)
+        this.saveContact(id).then(() => {
+          dplan.notify.notify('confirm', Translator.trans('confirm.saved'))
+        })
+      }
```

Comment:
This should work too and is less redundant
```suggestion
        const payload = {
          id: id === 'new' ? null : id,
          type: 'CustomerContact',
          attributes: {
            title: this.customerContact.title,
            phoneNumber: this.customerContact.phoneNumber ? this.customerContact.phoneNumber : null,
            text: this.customerContact.text ? this.customerContact.text : null,
            visible: this.customerContact.visible,
            eMailAddress: this.customerContact.eMailAddress ? this.customerContact.eMailAddress : null
          }
        }

      if (id === 'new') {
        this.createContact(payload).then((response) => {
          this.getContacts()
          dplan.notify.notify('confirm', Translator.trans('confirm.saved'))
        })
      } else {
        this.updateContact(payload)
        this.saveContact(id).then(() => {
          dplan.notify.notify('confirm', Translator.trans('confirm.saved'))
        })
      }
```

Reply:
you changed it? in which comment?